subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/configurations/DefaultConfigurationSpec.groovy

```diff
@@ -332,4 +335,61 @@ class DefaultConfigurationSpec extends Specification {
         dep.taskName == "bar"
         dep.configurationName == "conf"
     }
+
+    def "mutations are prohibited after resolution"() {
+        def listenerBroadcast = Mock(ListenerBroadcast) {
+            getSource() >> Mock(DependencyResolutionListener)
+        }
+        when:
+        def conf = new DefaultConfiguration(":conf", "conf", Mock(ConfigurationsProvider), resolver, listenerManager, Mock(DependencyMetaDataProvider), new DefaultResolutionStrategy())
```

Comment:
As I wrote in the commit message (6f08f1e97b122e4630dae3d2f06463d584fd4a36): I'm not sure if this test should be here, or if it should be made into an integration test.